File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        nums_dict = {}
        for index, number in enumerate(nums):
            second_number = target - number
            if second_number in nums_dict:
                return [nums_dict[second_number], index]
            nums_dict[number] = index

File: Data Structures & Algorithms/duplicate-integer/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
        // var existingNumbers = new Map();
        let existingNumbers = new Object();
        for (let i =0; i < nums.length; i++) {
            let eachNum = nums[i];
            console.log(existingNumbers[eachNum]);
            if (existingNumbers[eachNum]) {
                return true;
            } else {
                existingNumbers[eachNum] = true;
                console.log(existingNumbers)
            }
        };
        return false;
    }
}
